File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        tab = []
        l = 0
        r = 0
        le = 0
        maxl = 0
        while r < len(s):
            if s[r] not in tab:
                tab.append(s[r])
                r += 1
                le += 1
                maxl = max(maxl,le)

            else:
                tab.pop(0)
                l += 1
                
                le = len(tab)
        return maxl
                
        

File: Data Structures & Algorithms/house-robber-ii/submission-2.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int a = 0 , b = 0;
        int c = 0 , d = 0;
        if (nums.size() == 1){
            return nums[0];
        }
        for(int i = 0 ; i < nums.size()-1 ; i++){
            int temp = max( b , a+nums[i]);
            a = b;
            b = temp;
        }
        
        for(int i = 1 ; i < nums.size() ; i++){
            int temp = max( d , c+nums[i]);
            c = d;
            d = temp;
        }
        return max(b , d);
    }
};
